### PR TITLE
Player movement

### PR DESCRIPTION
**Pull request by sippeangelo**
This branch contains lots of collision code and other fixes that were required to get started with player movement, along with some new features listed below. Actual player movement is still far from done.
### CSpawner, CSpawnPoint

The Spawner component spawns an instance of an entity file at its position when receiving an ESpawnerSpawn event. If the Spawner has any children with the SpawnPoint component, it will select one at random to use as transform for the spawned entity.
### CTeam

The Team component is simply a flag that indicates which team a player belongs to. It should be used pretty much everywhere for team affiliation.
### CPlayerSpawn

PlayerSpawn is combined with a Team and a Spawner component to indicate a player spawn point. The PlayerSpawnSystem will call the Spawner when it receives the EInputCommand "PickTeam", where the value corresponds to the Team enum. It will then assign a Team component to the newly spawned entity with the same Team value as the Team component on the PlayerSpawn entity.
## Enums

Added enum support to entity file format. See CTeam for an example.
## Input binding format change

The old format of input bindings having a + or - in front of its command name was changed in favor of a system where the command value can be specified after a comma after the command name.
In other words:

``` ini
W=+Forward
S=-Forward
```

changed to

``` ini
W=Forward,1
S=Forward,-1
```
## Entity wrapper

Created `EntityWrapper` which wraps a `World` and an `EntityID` together to uniquely identify it across the game. It also contains some useful utility functions so you won't have to call functions directly on World all the time.

### CODE REVIEW

File: src/Game/Systems/PlayerSystem.cpp [hunk 1]
```diff
@@ -0,0 +1,42 @@
+#include "Systems/PlayerSystem.h"
+
+void PlayerSystem::UpdateComponent(World* world, EntityWrapper& entity, ComponentWrapper& component, double dt)
+{
+    component["Velocity"] = glm::vec3(0.f, 0.f, 0.f);
```
Comment (wima12) [inline src/Game/Systems/PlayerSystem.cpp]: This file could potentially be removed.
